src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -820,41 +820,44 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
             }
         }
 
-        private static TypedConstant GetParamArrayArgument(ParameterSymbol parameter, ImmutableArray<TypedConstant> constructorArgsArray, int argumentsCount, int argsConsumedCount, Conversions conversions)
+        private static TypedConstant GetParamArrayArgument(ParameterSymbol parameter, ImmutableArray<TypedConstant> constructorArgsArray,
+            ImmutableArray<string> constructorArgumentNamesOpt, int argumentsCount, int argsConsumedCount, Conversions conversions)
         {
             Debug.Assert(argsConsumedCount <= argumentsCount);
 
+            // If there's a named argument, we'll use that
+            if (!constructorArgumentNamesOpt.IsDefault)
+            {
+                int argIndex = constructorArgumentNamesOpt.IndexOf(parameter.Name);
+                if (TryGetNormalParamValue(parameter, constructorArgsArray, argIndex, conversions, out var namedValue))
```

Comment:
Does `TryGetNormalParamValue` handle `argIndex = -1`?

Reply:
There's indeed a problem there. Thanks
Fixed in new commit